### PR TITLE
Audio to text

### PR DESCRIPTION
**Pull request by yangcao77**
for issue https://issues.redhat.com/browse/DEVHAS-717

This PR pulls in audio-to-text sample and it's template to this repo. 
The audio-to-text template will only support `whisper.cpp` model, and therefore the script has been updated to disable the model selection list when generating the template

this PR also pull in rhoai fix to resolve the service & route name collision. 
Screenshot 2024-07-17 at 1 32 50 PM
Screenshot 2024-07-17 at 1 38 44 PM

Screenshot 2024-07-17 at 1 39 04 PM

### CODE REVIEW

File: templates/audio-to-text/template.yaml [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+          initContainer: quay.io/redhat-ai-dev/whisper-small:latest
+          appContainer: ${{ 'quay.io/redhat-ai-dev/ai-template-bootstrap-app:latest' if parameters.hostType === 'GitHub' else 'quay.io/redhat-ai-dev/audio-to-text:latest' }}
+          appPort: 8501
+          modelServiceContainer: quay.io/redhat-ai-dev/whispercpp:latest
+          modelServicePort: 8001
+          # for vllm
+          vllmSelected: ${{ parameters.modelServer === 'vLLM' }}
+          vllmModelServiceContainer: 
+          vllmModelName: 
+          maxModelLength:
```
Comment (maysunfaisal) [inline templates/audio-to-text/template.yaml:236]: can we remove these as well?

**Review by maysunfaisal — DISMISSED** [review]
small question, /lgtm

File: templates/audio-to-text/template.yaml (hunk 1 above)
Comment (yangcao77) [reply templates/audio-to-text/template.yaml]: updated

**Comment by yangcao77** [comment]
@maysunfaisal I've address your review comments. can you help review again please? Thanks

**Review by johnmcollier — APPROVED** [review]
Looks fine from my end

**Review by maysunfaisal — APPROVED** [review]

**Pull request merged**